TombEngine/Specific/Worker.cpp

```diff
@@ -0,0 +1,144 @@
+#include "framework.h"
+#include "Specific/Worker.h"
+
+namespace TEN::Utils
+{
+	WorkerManager g_Worker = WorkerManager();
+
+	WorkerManager::WorkerManager()
+	{
+		// Reserve threads.
+		unsigned int threadCount = std::thread::hardware_concurrency() * 2;
+		_threads.reserve(threadCount);
+
+		// Create threads.
+		for (int i = 0; i < threadCount; i++)
+			_threads.push_back(std::thread(&WorkerManager::Worker, this));
+
+		_deinitialize = false;
```

Comment:
Using operator= on an atomic variable induces std::memory_order_seq_cst which is probably not what you want here, especially when you consider that barring this access (which is not in parallel), all other accesses are under the protection of a mutex, which induces acquire-release ordering. You can probably let _deinitialize be a non atomic variable